src/components/hello_world.vue

```diff
@@ -1,61 +1,79 @@
-            :incoming_active_background_color="active_bkgrnd_color"
-            ref="switchy_toggle">
-      <template slot="on">
-        <p> Happy </p>
-      </template>
-      <template slot="off">
-        <p> <i class="fas fa-frown fa-lg"> </i></p>
-      </template>
-    </toggle>
-
-    <br>
-
-    <div class="icon-poc">
-      <i class="fas fa-check"></i>
-      <i class="far fa-check-circle"></i>
-      <i class="fas fa-angle-down"></i>
-    </div>
-
-    <br>
-    <view-file :incoming_filename="filename"
-               :incoming_file_content="content" ref="viewing_file"></view-file>
+    <button @click="$refs.modal1.open()">Open Default Modal</button>
+    <modal ref="modal1">
+      <h2>Title Example (Default)</h2>
+      <hr>
+      <p>The cool thing about this is that its totally reusable!</p>
+      <small>You can even close it using the little x at the top :D</small>
+    </modal>
+
+    <br/>
+    <br/>
+
+    <button @click="$refs.modal2.open()">Open Small Modal</button>
+    <modal ref="modal2" :size="'small'">
+      <h2>Title Example (Small)</h2>
+      <hr>
+      <p>The cool thing about this is that its totally reusable!</p>
+      <small>You can even close it using the little x at the top :D</small>
+    </modal>
+
+    <br/>
+    <br/>
+
+    <button @click="$refs.modal3.open()">Open Medium Modal</button>
+    <modal ref="modal3" :size="'medium'">
+      <h2>Title Example (Medium)</h2>
+      <hr>
+      <p>The cool thing about this is that its totally reusable!</p>
+      <small>You can even close it using the little x at the top :D</small>
+    </modal>
+
+    <br/>
+    <br/>
+
+    <button @click="$refs.modal4.open()">Open Large Modal</button>
+    <modal ref="modal4" :size="'large'">
+      <h2>Title Example (Large)</h2>
+      <hr>
+      <p>The cool thing about this is that its totally reusable!</p>
+      <small>You can even close it using the little x at the top :D</small>
+    </modal>
+
+    <br/>
+    <br/>
+
+    <button @click="$refs.modal5.open()">Open Custom Size Modal</button>
+    <modal ref="modal5" :custom_width="'900px'">
+      <h2>Title Example (900px)</h2>
+      <hr>
+      <p>The cool thing about this is that its totally reusable!</p>
+      <small>You can even close it using the little x at the top :D</small>
+    </modal>
 
   </div>
 </template>
 
 <script lang="ts">
 import { Component, Prop, Vue } from 'vue-property-decorator';
 
-import Toggle from './toggle.vue';
-
-import ViewFile from './view_file.vue';
+import Modal from './modal.vue';
 
 @Component({
-  components: {Toggle, ViewFile}
+  components: { Modal }
 })
-
 export default class HelloWorld extends Vue {
   @Prop()
   msg!: string;
 
-  on = "Hello";
-
-  off = "Super Off";
-
-  filename = "ke$ha.cpp";
-
-  content = "#include \"macklemore.cpp\"\nblah\nblah\nblah\n5coffeeeeeeee\n";
-
-  switchy: boolean = true;
-
-  active_bkgrnd_color = {
-    'backgroundColor': 'hotpink'
+  $refs!: {
```

Comment:
The type-checker doesn't actually type-check expressions in templates (sad, I know), so you can get rid of this for now